Adding info about service accounts in architecture.adoc (#259)
rancher/fleet-product-docs: community-docs/v0.15/modules/ROOT/pages/explanations/architecture.adoc

An overview of the components and how they interact on a high level.

image::FleetComponents.svg[Static, 600]

== Service accounts

{product_name} creates and uses several Kubernetes service accounts to securely manage the GitOps lifecycle. Understanding these service accounts and their permissions helps you maintain a least-privilege security model.

=== Management cluster service accounts

When you install {product_name} on the upstream management cluster, {product_name} creates service accounts for its core controllers. As you register downstream clusters and add Git repositories, {product_name} dynamically creates additional service accounts with restricted permissions.

==== ++fleet-controller++ service account

Namespace: ++cattle-fleet-system++

The ++fleet-controller++ service account is used by the main Fleet controller pod. It manages bundles, clusters, and cluster groups, reconciles {product_name} resources, and runs GitJobs to process Git repositories.

==== ++gitjob++ service account

Namespace: ++cattle-fleet-system++

> The ++gitjob++ service account is used by the GitJob controller. It creates temporary pods to clone Git repositories and generate bundles.

Reviewer: This phrasing may be misleading: git jobs themselves are not run using the `gitjob` service account.
The `gitjob` service account is used to run the `gitjob` controller, and the [job](https://github.com/rancher/fleet/blob/main/charts/fleet/templates/job_cleanup_gitrepojobs.yaml#L19) cleaning up git jobs.

However, those git jobs themselves are not run using that service account, but by another service account [created](https://github.com/rancher/fleet/blob/main/internal/cmd/controller/gitops/reconciler/rbac.go#L19) specifically for the git job, and named `git-<gitrepo_name>`.

==== ++import++ service account

> Namespace: ++fleet-default++

Reviewer: nit: We could explain that its name is random, built from a cluster registration token.

The ++import++ service account is created temporarily during downstream cluster registration.

{product_name} only grants permissions to:

* create ++ClusterRegistration++ requests
* read secrets within the ++cattle-fleet-clusters-system++ namespace

> After cluster registration completes, the downstream agent removes the account and no longer uses it.

Reviewer: ```suggestion
After cluster registration completes, the downstream agent removes the service account and no longer uses it.
```

Author: This is fixed.

==== ++request++ service account

Namespace: ++fleet-default++

The ++request++ service account becomes the permanent identity of a registered downstream cluster.

{product_name} restricts this account using least-privilege RBAC policies. The account can:

* list ++BundleDeployment++ resources within its assigned cluster namespace

> * list ++Content++ resources within its assigned cluster namespace

Reviewer: Content resources are cluster-scoped, not namespaced ;)
```suggestion
* list ++Content++ resources
```

Author: This is fixed now

* update the status subresources of ++BundleDeployment++ and ++Cluster++ resources

The account cannot:

* access data from other clusters
* modify desired state resources
* manage resources outside its assigned namespace

==== ++GitRepo++ service accounts

Namespace: ++fleet-default++

{product_name} automatically creates a dedicated service account, role, and role binding for each ++GitRepo++ resource. This isolation helps separate permissions between Git repositories and reduces cross-repository access.

=== Downstream cluster service accounts

Downstream clusters run {product_name} agents that retrieve configuration from the management cluster and perform Helm deployments.

==== ++fleet-agent++ service account

Namespace: ++cattle-fleet-system++

> The ++fleet-agent++ service account is used by the Fleet agent running on downstream clusters. It communicates with the upstream management cluster, monitors for new ++BundleDeployment++ resources, and reports deployment status changes.

Reviewer: The service account itself doesn't perform these operations, but it makes them possible. Perhaps we could rephrase this as something like:
```suggestion
The ++fleet-agent++ service account is used by the Fleet agent running on downstream clusters. It enables the agent to communicate with the upstream management cluster, to monitor its corresponding cluster namespace for new ++BundleDeployment++ resources in the upstream cluster, and to report deployment status changes.
```

The agent authenticates using the credentials associated with the ++request++ service account.

==== Deployment service accounts

When {product_name} deploys workloads to a downstream cluster, it uses a deployment service account to execute the deployment.

You can explicitly configure which downstream service account {product_name} should use by setting the ++serviceAccount++ field in:

* ++fleet.yaml++
* ++GitRepo++ resources

All service accounts referenced by a ++GitRepo++ must exist in the ++cattle-fleet-system++ namespace on the downstream cluster.

==== Restrict allowed service accounts

In multi-tenant environments, restrict which service accounts users can reference in Git repositories.

> Use a ++GitRepoRestriction++ resource to define an ++allowedServiceAccounts++ allowlist. If a ++GitRepo++ references an unauthorized service account, {product_name} blocks the deployment.

Reviewer: This is where `Policy` resources could be mentioned once #266 is merged.
